core/core-api/src/main/java/com/nerd/favorite18/core/api/ranking/service/RankRedisServiceV2.java

```diff
@@ -0,0 +1,66 @@
+package com.nerd.favorite18.core.api.ranking.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nerd.favorite18.core.api._common.support.error.CoreApiException;
+import com.nerd.favorite18.core.api._common.support.error.ErrorType;
+import com.nerd.favorite18.core.api.ranking.dto.SongRankDto;
+import com.nerd.favorite18.core.api.ranking.repository.RankRedisRepository;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.data.redis.core.ZSetOperations.TypedTuple;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+
+@RequiredArgsConstructor
+@Service
+public class RankRedisServiceV2 {
+    private final RankRedisRepository rankRedisRepository;
+
+    private final ObjectMapper objectMapper;
+
+    public void addSong(SongRankDto songRankDto) {
+        try {
+            String jsonValue = objectMapper.writeValueAsString(songRankDto);
+
+            rankRedisRepository.zAddSong(songRankDto.getSongId().toString(), jsonValue);
+        } catch (JsonProcessingException e) {
+            throw new CoreApiException(ErrorType.DEFAULT_ERROR, "Json processing failed in zAddSong");
+        }
+    }
+
+    public Double getScore(Long songId) {
+        return rankRedisRepository.zGetScore(String.valueOf(songId));
+    }
+
+    public void addScore(Long songId, Double score) {
+        rankRedisRepository.zAddScore(String.valueOf(songId), score);
+    }
+
+    public SongRankDto findSongById(String songId) {
+        String jsonValue = rankRedisRepository.zGetValue(songId);
+
+        if (!ObjectUtils.isEmpty(jsonValue)) {
+            try {
+                return objectMapper.readValue(jsonValue, SongRankDto.class);
+            } catch (JsonProcessingException e) {
+                throw new CoreApiException(ErrorType.DEFAULT_ERROR, "Json processing failed in findSongById");
+            }
+        }
+
+        return null;
+    }
```

Comment:
이건 선택의 문제인데 저는 반대로 해봤을것 같습니다 :)
```suggestion
        if (ObjectUtils.isEmpty(jsonValue)) {
            return null; // Optional을 고려해봤을듯 합니다. :)
        }

        try {
            return objectMapper.readValue(jsonValue, SongRankDto.class);
        } catch (JsonProcessingException e) {
            throw new CoreApiException(ErrorType.DEFAULT_ERROR, "Json processing failed in findSongById");
        }
    }
```

Reply:
optional 로 한 번 반영해 봤습니다. :)